### PR TITLE
AMGX - Initial recipe for v2.1.0. 

### PR DESCRIPTION
**Pull request by KristofferC**
https://github.com/NVIDIA/AMGX

The latest released version seems to only support CUDA 10 and earlier.

cc @maleadt

### CODE REVIEW

File: A/AMGX/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+products = [
+    LibraryProduct("libamgxsh", :libamgxsh),
+]
+
+dependencies = [
+    BuildDependency(PackageSpec(name = "CUDA_full_jll", version = "10.0"))
+]
+
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)
```
Comment (giordano) [inline A/AMGX/build_tarballs.jl:51]: GLIBCXX is in GCC 5: https://gcc.gnu.org/onlinedocs/libstdc++/manual/abi.html
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version=v"5.1")
```

**Comment by KristofferC** [comment]
Note, need to add a manual dependency of `CUDA_jll` to this jll with compact `CUDA = "~10.0"`.

**Comment by KristofferC** [comment]
@giordano What's the next steps here (with respect to having to add `CUDA_jll` as a dependency with specific compat). Do we merge this and then I just hand fix the jll or should i add `CUDA_jll` as a `Dependency` in th recipe (even though it isn't used in the build process)?

**Comment by giordano** [comment]
You can add `Dependency(PackageSpec(; name="CUDA_jll", uuid="...", ...))` (including the version compat, even though https://github.com/JuliaPackaging/BinaryBuilderBase.jl/issues/89 is an issue), the only problem I can see is that you'll get tons of warnings because of conflicting files between `CUDA_jll` and `CUDA_full_jll`.  I guess I found a use-case for a `RuntimeDependency` type.

**Comment by KristofferC** [comment]
I want to get `"~10.0"` as compat for `CUDA_jll`. How would I write that?

**Comment by giordano** [comment]
I guess that's the crux of https://github.com/JuliaPackaging/BinaryBuilderBase.jl/issues/89 :stuck_out_tongue:

**Comment by KristofferC** [comment]
So can I just fix it manually in the jll then?

**Comment by giordano** [comment]
Or directly in the registry PR, yes

**Comment by KristofferC** [comment]
So this is good to go now?

**Comment by maleadt** [comment]
> the only problem I can see is that you'll get tons of warnings because of conflicting files between `CUDA_jll` and `CUDA_full_jll`. I guess I found a use-case for a `RuntimeDependency` type.

CUDA_full_jll installs in `prefix/cuda`, whereas CUDA_jll directly puts stuff in the prefix. Still, at some point the CUDA_jll will `dlopen` stuff (and fail on `amdci`) so a RuntimeDependency would be valuable.

**Review by maleadt — APPROVED** [review]

**Pull request merged**